src/functional/geometry/convex_hull.zig

```diff
@@ -0,0 +1,243 @@
+    const allocator = testing.allocator;
+
+    var upper_hull = try PointSet.init(&allocator, 2);
+    defer upper_hull.deinit();
+    var lower_hull = try PointSet.init(&allocator, 2);
+    defer lower_hull.deinit();
+
+    var point: Point = Point{ .time = 0, .value = 3 };
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 1;
+    point.value = 2;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 2;
+    point.value = 3.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 3;
+    point.value = 5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 4;
+    point.value = 3;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 5;
+    point.value = 4;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 6;
+    point.value = 4;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 7;
+    point.value = 3;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 8;
+    point.value = 4.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 9;
+    point.value = 3.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 10;
+    point.value = 2.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 11;
+    point.value = 2.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 12;
+    point.value = 3.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 13;
+    point.value = 2.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 14;
+    point.value = 2.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 15;
+    point.value = 2.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 16;
+    point.value = 3;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 17;
+    point.value = 3;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 18;
+    point.value = 3;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 19;
+    point.value = 3;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 20;
+    point.value = 2.8;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+
+    try testing.expectEqual(5, upper_hull.len);
+    try testing.expectEqual(4, lower_hull.len);
+
+    // Expected Upper Hull.
+    try testing.expectEqual(0, upper_hull.points[0].time);
+    try testing.expectEqual(3, upper_hull.points[1].time);
+    try testing.expectEqual(8, upper_hull.points[2].time);
+    try testing.expectEqual(19, upper_hull.points[3].time);
+    try testing.expectEqual(20, upper_hull.points[4].time);
+    // Expected Lower Hull.
```

Comment:
Maybe add an empty line to make it clear two different things are checked.

Reply:
Fixed.